plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/SWTBrowserRegistry.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.registry;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.ModelPreferences;
+import org.jkiss.dbeaver.model.preferences.DBPPreferenceStore;
+import org.jkiss.dbeaver.utils.RuntimeUtils;
+import org.jkiss.utils.CommonUtils;
+
+import java.util.Arrays;
+
+
+public class SWTBrowserRegistry {
+
+    private SWTBrowserRegistry() {
+
+    }
+
+    private static final String INTERNAL_BROWSER_PROPERTY = "org.eclipse.swt.browser.DefaultType";
+
+    // Windows versions which could not have Edge as an available browser, or need to install it manually
+    private static final String[] LEGACY_OS = {
+        "Windows 98", //$NON-NLS-1$
+        "Windows XP", //$NON-NLS-1$
+        "Windows Vista", //$NON-NLS-1$
+        "Windows 7", //$NON-NLS-1$
+        "Windows 8", //$NON-NLS-1$
+        "Windows 8.1", //$NON-NLS-1$
+    };
+
+    public enum BrowserSelection {
+        EDGE("Microsoft Edge"),
+        IE("Internet Explorer");
+
+        private final String name;
+
+        BrowserSelection(@NotNull String name) {
+            this.name = name;
+        }
+
+        @NotNull
+        public String getFullName() {
+            return name;
+        }
+    }
+
+    /**
+     * @return Get selected browser or default browser if none is selected
+     */
```

Comment:
1. Consider leaving only summary and make it self explanatory so no `@return` is needed
2. Nullability annotations are missing